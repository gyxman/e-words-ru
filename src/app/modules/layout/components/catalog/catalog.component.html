<div class="list">
    <div
        *ngFor="let item of items; let i = index; trackBy: identityTrackByFunction"
        class="item"
    >
        <div
            class="item-content"
            [attr.automation-id]="'item-' + i"
            (click)="onClick(item)"
        >
            <h3 class="name">{{ item.name }}</h3>
        </div>
    </div>
</div>
