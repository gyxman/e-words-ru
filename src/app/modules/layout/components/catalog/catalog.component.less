@import '../../../../../styles/index';

:host {
    display: block;
}

.list {
    display: flex;
    flex-wrap: wrap;
    margin: -15px;
}

.item {
    width: 50%;
    padding: 15px;
    box-sizing: border-box;
}

.item-content {
    padding: 24px;
    border-radius: @radius-xl;
    background-color: @color-white;
    box-shadow: 0 0 16px 0 fade(@color-black, 24%);
    cursor: pointer;
}

.name {
    .font-h3();
    margin: 0;
}
