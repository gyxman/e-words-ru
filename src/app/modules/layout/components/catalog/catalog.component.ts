import {ChangeDetectionStrategy, Component} from '@angular/core';
import {CatalogItem} from '../../models/catalog-item';
import {Router} from '@angular/router';
import {LayoutRouteEnum} from '../../enums/layout-route.enum';
import {RouteEnum} from '../../../../enums/route.enum';
import {ComponentEnum} from '../../../exercises/enums/component.enum';

const items: CatalogItem[] = [
    {name: 'С английского на русский', component: ComponentEnum.InputRussian},
];

@Component({
    selector: 'app-catalog',
    templateUrl: './catalog.component.html',
    styleUrls: ['./catalog.component.less'],
    changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CatalogComponent {
    readonly identityTrackByFunction = (_index: number, item: any) => item;

    constructor(private router: Router) {}

    items = items;

    onClick({component}: CatalogItem) {
        this.router.navigate([RouteEnum.User, LayoutRouteEnum.Start], {
            queryParams: {component},
        });
    }
}
