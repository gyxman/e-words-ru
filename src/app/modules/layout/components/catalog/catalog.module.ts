import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {CatalogComponent} from './catalog.component';

@NgModule({
    imports: [CommonModule],
    declarations: [CatalogComponent],
    exports: [CatalogComponent],
})
export class CatalogModule {}
