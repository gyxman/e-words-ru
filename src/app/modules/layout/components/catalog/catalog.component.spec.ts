import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {CatalogComponent} from './catalog.component';
import {PageObject} from '../../../../testing/page-object';
import {Router} from '@angular/router';
import {deepEqual, instance, mock, verify} from 'ts-mockito';

describe('CatalogComponent - каталог упражнений', () => {
    let component: CatalogComponent;
    let fixture: ComponentFixture<CatalogComponent>;
    let pageObject: PageObject<CatalogComponent>;
    let routerMock: Router;

    beforeEach(() => {
        routerMock = mock(Router);
    });

    beforeEach(async(() => {
        TestBed.configureTestingModule({
            declarations: [CatalogComponent],
            providers: [
                {
                    provide: Router,
                    useFactory: () => instance(routerMock),
                },
            ],
        }).compileComponents();
    }));

    beforeEach(() => {
        fixture = TestBed.createComponent(CatalogComponent);
        component = fixture.componentInstance;
        pageObject = new PageObject(fixture);
    });

    it('Если компонент загрузился, то все элементы корректно отрисовались', () => {
        // act
        fixture.detectChanges();

        // assert
        expect(fixture.nativeElement).toMatchSnapshot();
    });

    it('Если пользователь нажимает на упражнение, выполняем переход к упражнению', () => {
        // arrange
        fixture.detectChanges();

        // act
        pageObject.click(pageObject.getByAutomationId('item-0'));

        // assert
        verify(
            routerMock.navigate(
                deepEqual(['user', 'start']),
                deepEqual({queryParams: {component: 'input-russian'}}),
            ),
        ).once();
    });
});
