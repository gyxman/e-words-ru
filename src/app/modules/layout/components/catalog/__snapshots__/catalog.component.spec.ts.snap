// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogComponent - каталог упражнений Если компонент загрузился, то все элементы корректно отрисовались 1`] = `
<div
  id="root0"
>
  <div
    class="list"
  >
    
    <div
      class="item"
    >
      <div
        automation-id="item-0"
        class="item-content"
      >
        <h3
          class="name"
        >
          С английского на русский
        </h3>
      </div>
    </div>
  </div>
</div>
`;
